### DISCUSSION TITLE
S905X3  X96 Max+ 4GB  boot problems, getting started

### DISCUSSION THREAD

**Discussion by NickelNuts** [devmfc/debian-on-amlogic#94] [Q&A] [answered]
I was unable to get  Armbian_community_24.11.0-trunk.66_Aml-s9xx-box_bookworm_current_6.6.47_minimal to boot on my TV-Box (X96Max+ 100 vA 4GBRAM).  I tried using the meson-sm1-x96-air.dtb because this is the model I use for Coreelec.  This did not boot at all, the system just boots to Android.

I also tried  this image “Devmfc_Debian-Bookworm_6.6.49-meson64_Minimal-24.09.05.img” from this project.  It seemed to be stuck in a boot loop with a blue screen on HDMI.

Going back to Armbian, I tried copying the file “meson-sm1-x96aira100.dtb”  from the into the Armbian boot folder and configured “extlinux/extlinux.conf “ to look for the correct dtb.  I renamed “u-boot-s905x3” to “u-boot.ext”.  This did not boot.

I copied the contents of the entire boot folder from devmfc Debian Bookworm (v12) with Linux kernel 6.6.49 to the boot folder of Armbian Linux v6.6, overwriting "aml_autoscript", "emmc_autoscript", and  "s905_autoscript".    I edited “boot.config” to use “box=x96aira100” and set the variable “rd_img=uInitrd”.  This boots into Armbian successfully enough that I get a command prompt on the HDMI monitor, and I get an IP address on the Ethernet port.  After logging in by ssh and creating an Arbian account, I did uname to check the kernel version: “root@aml-s9xx-box:~# uname -a
Linux aml-s9xx-box 6.6.49-meson64 #19 SMP PREEMPT Thu Sep  5 00:58:55 CEST 2024 aarch64 GNU/Linux”

Of course I can’t update anything without breaking the system, and this kernel is lacking a lot of features I would like to use on a desktop system.  I know Armbian is off-topic here, so I will post on the Armbian site about how to try to get that working for me, but I thought I'd leave this here in case it is helpful somehow in developing a working system for this box.

**Comment by devmfc** [comment, marked as answer]
X96 Air P3 and X96 Max + boxes are my main development targets. They will definitely boot with these images.

It is best to use box=x96aira100 for your box if it is a 100Mb/s box.

If your box doesn't boot there are three possible problems with your box:
1. You don't have stock u-boot bootloader (so you installed some other image like Slimbox or something). Revert to original android image first
2. You are booting from a incompatible SD CARD, try a USB flash disk
3. You are using a poor quality power supply

I think these boxes are too slow for desktop usage, that's right, but there are AFAIK no features missing to install the default desktop environments. So What feature are you missing in these kernels for desktop usage?

**Comment by NickelNuts** [comment]
It was #1; I do have Slimbox on there.  Your image booted right up after I copied the file "u-boot-s905x3" from the Armbian boot folder into my SD card root partition, and renamed it "u-boot.ext". The missing feature question is some confusion I have around hardware acceleration, and whether Lima is enabled in the kernel?  I am using your build on an X96 mini box with S905W CPU, and it is great for a little server. So, I am glad to have this going.  I prefer it to Armbian.  This is really cool having the temperature on the display  EDIT:  Nevermind about the graphics question, I see panfrost in the boot log, and glxgears gives 60fps
